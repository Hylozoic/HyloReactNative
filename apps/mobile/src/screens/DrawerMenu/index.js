import component from './DrawerMenu'

export default component
