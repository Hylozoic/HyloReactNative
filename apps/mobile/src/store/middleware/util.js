export function isPromise (value) {
  return value && typeof value.then === 'function'
}
