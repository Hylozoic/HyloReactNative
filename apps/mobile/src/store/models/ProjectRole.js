import { attr, fk, Model } from 'redux-orm'

class ProjectRole extends Model {
  toString () {
    return `Attachment (${this.type}): ${this.url}`
  }
}

export default ProjectRole

ProjectRole.modelName = 'ProjectRole'

ProjectRole.fields = {
  id: attr(),
  name: attr(),
  project: fk('Post', 'attachments')
}
