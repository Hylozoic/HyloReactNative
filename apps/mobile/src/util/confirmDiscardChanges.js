import { useTranslation } from 'react-i18next'
import { Alert } from 'react-native'

export default function confirmDiscardChanges ({
  onDiscard,
  hasChanges = true,
  title = 'You have unsaved changes',
  confirmationMessage = 'Are you sure you want to discard your changes?',
  disgardButtonText = 'Discard',
  continueButtonText = 'Continue Editing',
  t
}) {
  t('You have unsaved changes')
  t('Are you sure you want to discard your changes?')
  t('Discard')
  t('Continue Editing')
  if (hasChanges) {
    Alert.alert(
      t(title),
      t(confirmationMessage),
      [
        { text: t(disgardButtonText), onPress: onDiscard },
        { text: t(continueButtonText), style: 'cancel' }
      ])
  } else {
    onDiscard()
  }
}
