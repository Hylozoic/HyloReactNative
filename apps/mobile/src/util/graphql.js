import gql from 'graphql-tag'
import { print } from 'graphql/language/printer'

export function stringToGraphql (graphqlString) {
  return (typeof graphqlString === 'string' || graphqlString instanceof String)
    ? gql(graphqlString)
    : graphqlString
}

export function graphqlToString (unknownGraphql) {
  // Enables blended use of strings and gql tags for graphql queries
  return (typeof unknownGraphql === 'object' && !(unknownGraphql instanceof String))
    ? print(unknownGraphql)
    : unknownGraphql
}
