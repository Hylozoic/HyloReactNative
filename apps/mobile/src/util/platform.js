import { Platform } from 'react-native'
import DeviceInfo from 'react-native-device-info'

export const isIOS = Platform.OS === 'ios'
export const appVersion = DeviceInfo.getVersion()
export const platformName = isIOS ? 'ios' : 'android'
