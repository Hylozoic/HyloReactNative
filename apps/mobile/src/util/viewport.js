import { Dimensions } from 'react-native'
const dimensions = Dimensions.get('window')

// TODO: figure out what to do about rotation
export const vw = dimensions.width / 100
export const vh = dimensions.height / 100
