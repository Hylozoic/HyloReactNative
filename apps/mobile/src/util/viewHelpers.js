import { isIOS } from './platform'

export const keyboardAvoidingViewProps = {
  behavior: 'padding',
  keyboardVerticalOffset: isIOS ? 64 : 80
}
