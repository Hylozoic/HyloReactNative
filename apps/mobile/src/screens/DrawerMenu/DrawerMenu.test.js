import 'react-native'
import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import TestRenderer from 'react-test-renderer'
import DrawerMenu, { SectionHeader, NetworkRow, GroupRow, TextButton } from './DrawerMenu'
import { ALL_GROUP } from 'store/models/Group'
import { TestRoot } from 'util/testing'

// jest.mock('react-native', () => ({
//   ...jest.requireActual('react-native'),
//   BackHandler: { addEventListener: jest.fn(), mockOnBackPress: jest.fn() }
// }))

describe('DrawerMenu', () => {
  const minProps = {
    name: 'Roy Rogers',
    avatarUrl: 'http://anyurl',
    memberships: [],
    goToGroup: () => {},
    goToMyProfile: () => {},
    showSettings: () => {},
    currentGroupId: 12,
    networks: [],
    goToGroupSettings: jest.fn(),
    currentGroup: { name: 'foo', avatarUrl: 'someurl' },
    groups: []
  }

  it('matches the last snapshot', () => {
    const props = {
      networks: [{
        id: 1,
        name: 'Network'
      }],
      groups: [{
        id: 2,
        name: 'Group'
      }]
    }
    const renderer = new ReactShallowRenderer()
    // const { toJSON } = TestRenderer.create(
    renderer.render(
      <TestRoot>
        <DrawerMenu {...minProps} {...props} />
      </TestRoot>
    )

    // expect(toJSON()).toMatchSnapshot()
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})

describe('SectionHeader', () => {
  it('matches the last snapshot', () => {
    const renderer = new ReactShallowRenderer()
    renderer.render(<SectionHeader section={{ label: 'Networked Groups' }} />)
    const actual = renderer.getRenderOutput()

    expect(actual).toMatchSnapshot()
  })
})

describe('GroupRow', () => {
  it('matches the last snapshot', () => {
    const renderer = new ReactShallowRenderer()
    const group = {
      id: 1,
      avatarUrl: 'foo.png',
      name: 'Foom',
      newPostCount: 7
    }
    renderer.render(<GroupRow
      group={group}
      goToGroup={() => {}}
      currentGroupId={2}
                    />)
    const actual = renderer.getRenderOutput()

    expect(actual).toMatchSnapshot()
  })

  it('hides badge when no newPostCount', () => {
    const renderer = new ReactShallowRenderer()
    const group = {
      id: 1,
      avatarUrl: 'foo.png',
      name: 'Foom'
    }
    renderer.render(<GroupRow
      group={group}
      goToGroup={() => {}}
      currentGroupId={2}
                    />)
    const actual = renderer.getRenderOutput()

    expect(actual).toMatchSnapshot()
  })

  it('highlights when matching currentGroupId', () => {
    const renderer = new ReactShallowRenderer()
    const group = {
      id: 1,
      avatarUrl: 'foo.png',
      name: 'Foom'
    }
    renderer.render(<GroupRow
      group={group}
      goToGroup={() => {}}
      currentGroupId={group.id}
                    />)
    const actual = renderer.getRenderOutput()

    expect(actual).toMatchSnapshot()
  })
})

it('TextButton renders', () => {
  const renderer = new ReactShallowRenderer()
  renderer.render(<TextButton text='anything' onPress={() => {}} />)
  const actual = renderer.getRenderOutput()

  expect(actual).toMatchSnapshot()
})
