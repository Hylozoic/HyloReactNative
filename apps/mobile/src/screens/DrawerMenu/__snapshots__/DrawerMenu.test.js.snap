// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMenu matches the last snapshot 1`] = `
<SafeAreaProvider>
  <TRenderEngineProvider
    baseStyle={
      {
        "fontSize": 14,
      }
    }
    classesStyles={{}}
    customHTMLElementModels={{}}
    emSize={14}
    enableCSSInlineProcessing={true}
    enableUserAgentStyles={true}
    fallbackFonts={
      {
        "monospace": "Menlo",
        "sans-serif": "system",
        "serif": "Times New Roman",
      }
    }
    htmlParserOptions={
      {
        "decodeEntities": true,
      }
    }
    ignoredDomTags={[]}
    ignoredStyles={[]}
    systemFonts={
      [
        "San Francisco",
        "Arial",
        "ArialHebrew",
        "Avenir",
        "Baskerville",
        "Bodoni 72",
        "Bradley Hand",
        "Chalkboard SE",
        "Cochin",
        "Copperplate",
        "Courier",
        "Courier New",
        "Damascus",
        "Didot",
        "Futura",
        "Geeza Pro",
        "Georgia",
        "Gill Sans",
        "Helvetica",
        "Helvetica Neue",
        "Hiragino Sans",
        "Hoefler Text",
        "Iowan Old Style",
        "Kailasa",
        "Khmer Sangam MN",
        "Marker Felt",
        "Menlo",
        "Mishafi",
        "Noteworthy",
        "Optima",
        "Palatino",
        "Papyrus",
        "Savoye LET",
        "Symbol",
        "Thonburi",
        "Times New Roman",
        "Trebuchet MS",
        "Verdana",
        "Zapf Dingbats",
        "Zapfino",
      ]
    }
    tagsStyles={{}}
  >
    <Provider
      store={
        {
          "dispatch": [MockFunction],
          "getState": [MockFunction],
          "subscribe": [MockFunction],
        }
      }
    >
      <HyloHTMLConfigProvider>
        <ForwardRef(NavigationContainerInner)>
          <DrawerMenu
            avatarUrl="http://anyurl"
            currentGroup={
              {
                "avatarUrl": "someurl",
                "name": "foo",
              }
            }
            currentGroupId={12}
            goToGroup={[Function]}
            goToGroupSettings={[MockFunction]}
            goToMyProfile={[Function]}
            groups={
              [
                {
                  "id": 2,
                  "name": "Group",
                },
              ]
            }
            memberships={[]}
            name="Roy Rogers"
            networks={
              [
                {
                  "id": 1,
                  "name": "Network",
                },
              ]
            }
            showSettings={[Function]}
          />
        </ForwardRef(NavigationContainerInner)>
      </HyloHTMLConfigProvider>
    </Provider>
  </TRenderEngineProvider>
</SafeAreaProvider>
`;

exports[`GroupRow hides badge when no newPostCount 1`] = `
<View>
  <TouchableOpacity
    onPress={[Function]}
  >
    <FastImage
      source={
        {
          "uri": "foo.png",
        }
      }
    />
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
    >
      Foom
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`GroupRow highlights when matching currentGroupId 1`] = `
<View>
  <TouchableOpacity
    onPress={[Function]}
  >
    <FastImage
      source={
        {
          "uri": "foo.png",
        }
      }
    />
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
    >
      Foom
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`GroupRow matches the last snapshot 1`] = `
<View>
  <TouchableOpacity
    onPress={[Function]}
  >
    <FastImage
      source={
        {
          "uri": "foo.png",
        }
      }
    />
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
    >
      Foom
    </Text>
    <View>
      <Text>
        7
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`SectionHeader matches the last snapshot 1`] = `
<View>
  <Text>
    NETWORKED GROUPS
  </Text>
</View>
`;

exports[`TextButton renders 1`] = `
<TouchableOpacity
  hitSlop={
    {
      "bottom": 10,
      "left": 10,
      "right": 15,
      "top": 20,
    }
  }
  onPress={[Function]}
>
  <Text>
    anything
  </Text>
</TouchableOpacity>
`;
